src/workflows-app/components/FilterableWorkflowTable.ts

```diff
@@ -113,12 +162,22 @@ const FilterableWorkflowTable = ({
     Succeeded = 'Succeeded',
   }
 
-  const filteredPreviousRuns: Run[] = filterOption ? getFilteredRuns(filterOption, runsData) : runsData;
   const filterOptions: FilterOptions[] = [FilterOptions.Error, FilterOptions.NoFilter, FilterOptions.Succeeded];
+  const filteredPreviousRuns: Run[] = useMemo(
+    () => (filterOption ? getFilteredRuns(filterOption, runsData, errorStates) : runsData),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Ignoring this rule can often lead to subtle bugs. It would be nice to add a comment explaining why it's ignored.